atc/worker/pool.go

```diff
@@ -10,143 +10,112 @@ import (
 
 	"code.cloudfoundry.org/lager"
 	"code.cloudfoundry.org/lager/lagerctx"
-
 	"github.com/concourse/concourse/atc/db"
 	"github.com/concourse/concourse/atc/metric"
+	"github.com/concourse/concourse/atc/runtime"
+	"github.com/cppforlife/go-semi-semantic/version"
 	"github.com/hashicorp/go-multierror"
 )
 
-const WorkerPollingInterval = 5 * time.Second
+var PollingInterval = 5 * time.Second
 
-type NoCompatibleWorkersError struct {
-	Spec WorkerSpec
-}
+type Pool struct {
+	factory       Factory
+	db            DB
+	workerVersion version.Version
```

Comment:
That seems pretty reasonable to me. I wonder what the reason is for using the random placement strategy for checks - I had forgotten about that/hadn't thought too much about it at the time. It seems weird because then we'd be ignoring the e.g. `limit-active-containers/volumes` strategies and could knowingly place checks on a "bad" worker